spec/controllers/keywords_controller_spec.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe KeywordsController, type: :controller do
+  describe 'GET#new' do
+    context 'given the user_id session' do
+      it 'returns a successful response' do
+        user = Fabricate(:user)
+        session[:user_id] = user[:id]
+
+        get :new
+
+        expect(response).to be_successful
+      end
+
+      it 'renders the template of :new action' do
+        user = Fabricate(:user)
+        session[:user_id] = user[:id]
+
+        get :new
+
+        expect(response).to render_template(:new)
+      end
+    end
+
+    context 'given NO user_id session' do
+      it 'redirects to login' do
+        get :new
+
+        expect(response).to redirect_to(login_path)
+      end
+    end
+  end
+
+  describe 'POST#create' do
+    context 'given valid parameters (file)' do
+      it 'inserts keywords to database' do
+        user = Fabricate(:user)
+        session[:user_id] = user[:id]
+        file = fixture_file_upload('files/example.csv', 'text/csv')
+
+        expect do
+          post :create, params: { keyword: { file: file } }
+        end.to change(Keyword, :count).by(6)
+      end
+
+      it 'redirects to keywords path' do
+        user = Fabricate(:user)
+        session[:user_id] = user[:id]
+        file = fixture_file_upload('files/example.csv', 'text/csv')
+
+        post :create, params: { keyword: { file: file } }
+
+        expect(response).to redirect_to(keywords_path)
+      end
+
+      it 'shows a notice flash' do
```

Comment:
How about adding the test that creates keywords in the database here?

Reply:
Could you please explain more about this?

Reply:
I mean we can add the test to making sure that this controller will create some Keyword in the database by counting the Keyword from the database after this action has completed.

```
it 'creates the keywords' do
  user = Fabricate(:user)
  session[:user_id] = user[:id]
  file = fixture_file_upload('files/example.csv', 'text/csv')

  expect { post :create, params: { keyword: { file: file } } }.to change { Keyword.count }.from(0).to(xxxx)
end
```

Reply:
I think I have it here
https://github.com/gutakk/ruby-google-scraper/blob/feature%2Fupload-csv/spec/controllers/keywords_controller_spec.rb#L38-L46

Reply:
Oh my bad @gutakk